qutip/solver/heom/bofin_baths.py

```diff
@@ -337,6 +339,85 @@ def combine(cls, exponents, rtol=1e-5, atol=1e-7):
 
         return new_exponents
 
+    def spectral_density(self, w):
+        raise NotImplemented
+
+    def correlation_function(
+            self, t, epsabs=1e-6, epsrel=1e-6, method='gk15'):
+        """ Calculates the correlation function
+            by numerically computing the integral
+        Parameters
+        ----------
+        t : np.array or float
+            the time at which to evaluare the correlation function
+        epsabs : float
+            Absolute error tolerance
+        epsrel : float
+            Relative error tolerance
+        """
+        def c_i(w, t): return (1/np.pi)*self.spectral_density(w)*(
+            (1/np.tanh(w/(2*self.T)))*np.cos(w*t) - 1j*np.sin(w*t))
```

Comment:
does it make sense to also have a check for zerodivisionerror here, and if it catches it use np.sign() or something instead?